hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/JoinCheckOperation.java

```diff
@@ -61,12 +66,13 @@ public void run() {
             ILogger logger = getLogger();
             try {
                 if (service.getClusterJoinManager().validateJoinMessage(request)) {
-                    try {
-                        // also validate other node's version is able to join our cluster
-                        node.getNodeExtension().validateJoinRequest(request);
+                    // also validate other cluster's major.minor version is same as this cluster
```

Comment:
Isn't this what `node.getNodeExtension().validateJoinRequest(request)` will do anyway - check the cluster version? Why was that method call removed? Because it throws an exception instead of just returning false?